### ISSUE TITLE
losotoImporter problem for solset._f_get_child('antenna')

### ISSUE THREAD

**Issue by oonkjbr** [lofar-astron/prefactor#171]
Hey All, (espc. TJD and FdG),

For a Singularity image containing LOFAR 2_20_2 and Losoto version 1.0 (I know this is old, but it should work): 

I am trying to run prefactor (the clock-tec part on the calibrator solutions). 

Script losotoImporter.py (losoto exists and the import of h5parm, solWriter does NOT throw an error)

from losoto.h5parm import h5parm, solWriter
solset = h5parmDB.makeSolset(solsetName)
#ERROR IS HERE:solset._f_get_child('antenna')

The script we use is here:

https://github.com/apmechev/prefactor/blob/master/scripts/losotoImporter.py

Output log error:

018-05-02 15:50:45 WARNING genericpipeline.executable_args:   from _parmdb import ParmDB
2018-05-02 15:50:46 WARNING genericpipeline.executable_args: WARNING:root:No sky table found. (Direction-dependent parameters will not work.)
2018-05-02 15:52:15 WARNING genericpipeline.executable_args: ERROR:node.e0683.python_plugin:group ``/sol000`` does not have a child named ``_f_getChild``
2018-05-02 15:52:15 WARNING genericpipeline.executable_args: INFO:node.e0683.python_plugin:Total time 90.4106s; user time: 40.8550s; system time: 17.4730s
2018-05-02 15:52:15 WARNING genericpipeline.executable_args: DEBUG:node.e0683.python_plugin:Start time was 1525269045.1360s; end time was 1525269135.5469s
2018-05-02 15:52:15 DEBUG   genericpipeline.executable_args: Results for job 0 submitted by ('172.16.18.175', 57607)
2018-05-02 15:52:15 ERROR   node.e0683.python_plugin: group ``/sol000`` does not have a child named ``_f_getChild``
2018-05-02 15:52:15 INFO    node.e0683.python_plugin: Total time 90.4106s; user time: 40.8550s; system time: 17.4730s
2018-05-02 15:52:15 DEBUG   node.e0683.python_plugin: Start time was 1525269045.1360s; end time was 1525269135.5469s
2018-05-02 15:52:16 WARNING genericpipeline.executable_args: Closing remaining open files:./h5imp_cal_losoto.h5...done
2018-05-02 15:52:16 DEBUG   genericpipeline.executable_args:

2018-05-02 15:52:17 WARNING genericpipeline: pythonplugin reports failure (using executable_args recipe)
2018-05-02 15:52:17 ERROR   genericpipeline: *******************************************
2018-05-02 15:52:17 ERROR   genericpipeline: Failed pipeline run: Pre-Facet-Calibrator-2
2018-05-02 15:52:17 ERROR   genericpipeline: Detailed exception information:
2018-05-02 15:52:17 ERROR   genericpipeline: <class 'lofarpipe.support.lofarexceptions.PipelineRecipeFailed'>
2018-05-02 15:52:17 ERROR   genericpipeline: pythonplugin failed
2018-05-02 15:52:17 ERROR   genericpipeline: *******************************************
2018-05-02 15:52:17 ERROR   genericpipeline: LOFAR Pipeline finished unsuccesfully.
2018-05-02 15:52:17 WARNING genericpipeline: recipe genericpipeline completed with errors

**Comment by tammojan** [comment]
I don't know how this can ever have worked, the Solset object indeed does not have a method `._f_get_child`. Its member `solset.obj` does have this method, and it seems like this is what is meant in the code. Replacing `solset._f_get_child` by `solset.obj._f_get_child` should do the trick.

**Comment by oonkjbr** [comment]
Hi TJD,

Thanks for the suggestion, but I dont follow it. I have tried making the suggested changes, but it then crashed with an error: ERROR:node.e0256.python_plugin:group ``/sol000`` does not have a child named ``obj``.

So apparently the .obj is not recognized or does not exist. If I check line 309 in:
https://github.com/lofar-astron/prefactor/blob/master/scripts/losotoImporter.py

I also find:
antennaTable = solset._f_get_child('antenna')

Any ideas?

-R

See below for my full error:

CHANGES:
ketelmeer [264] > diff losotoImporter.py 03052018/losotoImporter.py 
302c302
<     antennaTable = solset.obj._f_get_child('antenna')
---
>     antennaTable = solset.obj._f_gcget_child('antenna')
310c310
<     sourceTable = solset.obj._f_get_child('source')
---
>     sourceTable = solset.obj._f_gcget_child('source')
316c316
<         c = solset.obj._f_getChild(tab)
---
>         c = solset.obj._f_gcgetChild(tab)


OUT LOG FROM NEW RUN
2018-05-03 12:39:33 WARNING genericpipeline.executable_args:   from _parmdb import ParmDB
2018-05-03 12:39:33 WARNING genericpipeline.executable_args: WARNING:root:No sky table found. (Direction-dependent parameters will not work.)
2018-05-03 12:41:01 WARNING genericpipeline.executable_args: ERROR:node.e0256.python_plugin:group ``/sol000`` does not have a child named ``obj``
2018-05-03 12:41:01 WARNING genericpipeline.executable_args: INFO:node.e0256.python_plugin:Total time 88.6716s; user time: 41.8570s; system time: 16.2840s
2018-05-03 12:41:01 WARNING genericpipeline.executable_args: DEBUG:node.e0256.python_plugin:Start time was 1525343972.5531s; end time was 1525344061.2251s
2018-05-03 12:41:01 ERROR   node.e0256.python_plugin: group ``/sol000`` does not have a child named ``obj``
2018-05-03 12:41:01 INFO    node.e0256.python_plugin: Total time 88.6716s; user time: 41.8570s; system time: 16.2840s
2018-05-03 12:41:01 DEBUG   node.e0256.python_plugin: Start time was 1525343972.5531s; end time was 1525344061.2251s
2018-05-03 12:41:01 DEBUG   genericpipeline.executable_args: Results for job 0 submitted by ('172.16.17.2', 38163)
2018-05-03 12:41:02 WARNING genericpipeline.executable_args: Closing remaining open files:./h5imp_cal_losoto.h5...done
2018-05-03 12:41:02 DEBUG   genericpipeline.executable_args: 
2018-05-03 12:41:02 WARNING genericpipeline.executable_args: 
2018-05-03 12:41:02 INFO    genericpipeline.executable_args: Subprocess completed with exit status 1: /bin/sh -c python /opt/lofar/lofar/lib64/python2.7/site-packages/lofarpipe/recipes/nodes/python_plugin.py 0 172.16.17.2 37011
2018-05-03 12:41:02 ERROR   genericpipeline.executable_args: Remote process python /opt/lofar/lofar/lib64/python2.7/site-packages/lofarpipe/recipes/nodes/python_plugin.py ["['/mnt/lustre/inula/shared/lofar/oonk/ln38s_pref_cal2_L632075_000...
